codeU2.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Jan 13 10:22:28 2017
+Exercise set 2, Code U
+@author: Luiza
+"""
+import logging
+import unittest
+
+class Tree():
+    
+    """Defines the class for the tree:
+    self.children contains a dict of pairs (node:(left_child,right_child))
+    self.head contains the value of the tree head
+    None denotes the absence if the node
+    """
+    
+    def __init__(self,children,head):
+        self.children = children;
+        self.head = head;
+
+def dfs_path(tree_, start, end, path=None):
```

Comment:
The `path=None` is never overridden anywhere so the argument should be removed for simplicity. If this was a `Tree` class method then `tree_` would be `self` and `start` could be omitted for `self.head` as well.